WordPress/Classes/Networking/RemoteBlogOptionsHelper.m

```diff
@@ -9,8 +9,7 @@ + (NSDictionary *)mapOptionsFromResponse:(NSDictionary *)response
 {
     NSMutableDictionary *options = [NSMutableDictionary dictionary];
     options[@"home_url"] = response[@"URL"];
-    // We'd be better off saving this as a BOOL property on Blog, but let's do what XML-RPC does for now
-    options[@"blog_public"] = [[response numberForKey:@"is_private"] boolValue] ? @"-1" : @"0";
+    options[@"blog_public"] = [response valueForKeyPath:@"options.blog_public"];
```

Comment:
Okay... actually two requests.  As a small optimization I think we could move the work being done here into the `if` block below and include `blog_public` in the `optionsDirectMapKeys` array.